src/main/kotlin/com/exactpro/th2/codec/EventProcessor.kt

```diff
@@ -0,0 +1,209 @@
+    protected abstract fun storeEachErrorEvent(errorEventId: String, message: String, events: Set<EventID>)
+
+    private fun List<MessageID>.joinToReadableString(): String =
+        joinToString(", ") {
+            "${it.connectionId.sessionAlias}:${it.direction}:${it.sequence}[.${it.subsequenceList.joinToString(".")}]"
+        }
+
+    companion object {
+        private val LOGGER = KotlinLogging.logger {}
+    }
+}
+
+class LogOnlyEventProcessor : AbstractEventProcessor() {
+    override fun storeEvent(
+        message: String,
+        messagesIds: List<MessageID>,
+        body: List<String>
+    ): String = DEFAULT_EVENT_ID
+
+    override fun storeErrorEvent(
+        message: String,
+        messagesIds: List<MessageID>,
+        cause: Throwable?,
+        additionalBody: List<String>
+    ) = DEFAULT_EVENT_ID
+
+    override fun storeEachEvent(warnEvent: String, message: String, events: Set<EventID>) {}
+    override fun storeEachErrorEvent(errorEventId: String, message: String, events: Set<EventID>) {}
+
+    companion object {
+        private const val DEFAULT_EVENT_ID = "0000"
+    }
+}
+
+class StoreEventProcessor(private val storeEventFunc: (Event, EventID?) -> Unit) : AbstractEventProcessor() {
+
+    override fun storeEvent(
+        message: String,
+        messagesIds: List<MessageID>,
+        body: List<String>
+    ): String {
+        val event = createEvent(message, messagesIds, body = body)
+        storeEventFunc(event, null)
+        return event.id
+    }
+
+    override fun storeErrorEvent(
+        message: String,
+        messagesIds: List<MessageID>,
+        cause: Throwable?,
+        additionalBody: List<String>
+    ): String {
+        val event = createEvent(message, messagesIds, FAILED, cause, additionalBody)
+        storeEventFunc(event, null)
+        return event.id
+    }
+
+    override fun storeEachEvent(
+        warnEvent: String,
+        message: String,
+        events: Set<EventID>
+    ) {
+        events.forEach {
+            it.addReferenceTo(warnEvent, message, PASSED)
+        }
+    }
+
+    override fun storeEachErrorEvent(
+        errorEventId: String,
+        message: String,
+        events: Set<EventID>
+    ) {
+        events.forEach {
+            it.addReferenceTo(errorEventId, message, FAILED)
+        }
+    }
+
+    private fun EventID?.addReferenceTo(eventId: String, name: String, status: Status): EventID {
+        Event.start()
+            .endTimestamp()
+            .name(name)
+            .status(status)
+            .type(if (status != PASSED) "Error" else "Warn")
+            .bodyData(EventUtils.createMessageBean("This event contains reference to the codec event"))
+            .bodyData(ReferenceToEvent(eventId))
+            .also { event ->
+                storeEventFunc(event, this)
+            }.run {
+                return checkNotNull(EventUtils.toEventID(startTimestamp, "", "", id))
```

Comment:
Sorry, but what does happen here exactly? Why do we create a fake event ID without book and scope? These values should be taken from the parent event ID